app/modules/inventory/lib/scanner/quagga.ts

```diff
@@ -0,0 +1,104 @@
+}
+
+export async function startQuaggaScanner (params: ScannerParams) {
+  // For some unknown reason, importing Quagga statically makes scanning fail:
+  //   message: 't is undefined',
+  //   stack: o.createLiveStream@webpack-internal:///./node_modules/quagga/dist/quagga.min.js:1:44717
+  //     i@webpack-internal:///./node_modules/quagga/dist/quagga.min.js:1:24184
+  //     init@webpack-internal:///./node_modules/quagga/dist/quagga.min.js:1:29451
+  //     startScanning/<@webpack-internal:///./app/modules/inventory/lib/scanner/embedded.js:46:41
+  // But importing it dynamically works
+  const { default: Quagga } = await import('@ericblade/quagga2')
+
+  const { containerElement, beforeScannerStart, onDetectedActions, setStopScannerCallback } = params
+  assert_.object(containerElement)
+  // Using a promise to get a friendly way to pass errors
+  // but this promise will never resolve, and will be terminated,
+  // if everything goes well, by a cancel event
+  return new Promise((resolve, reject) => {
+    log_.info('starting quagga initialization')
+
+    let cancelled = false
+    const stopScanner = function () {
+      log_.info('stopping quagga')
+      cancelled = true
+      return Quagga.stop()
+    }
+
+    setStopScannerCallback(stopScanner)
+
+    const config = getConfig(containerElement)
+
+    Quagga.init(config, err => {
+      if (cancelled) return
+      if (err) {
+        err.reason = 'permission_denied'
+        return reject(err)
+      }
+
+      beforeScannerStart?.()
+      log_.info('quagga initialization finished. Starting')
+      Quagga.start()
+
+      Quagga.onProcessed(drawCanvasFactory(Quagga))
+
+      Quagga.onDetected(onDetectedFactory(onDetectedActions))
+    })
+  })
+}
+
+const verticalMargin = '30%'
+const horizontalMargin = '15%'
+
+// See doc: https://github.com/ericblade/quagga2#quaggainitconfig-callback
+function getConfig (containerElement: HTMLElement) {
+  return {
+    inputStream: {
+      name: 'Live',
+      type: 'LiveStream' as InputStreamType,
+      area: {
+        top: verticalMargin,
+        right: horizontalMargin,
+        left: horizontalMargin,
+        bottom: verticalMargin,
+      },
+      target: containerElement,
+      constraints: getConstraints(),
+    },
+    // From the docs https://github.com/ericblade/quagga2?tab=readme-ov-file#locate:
+    // "Why would someone turn this feature off? Localizing a barcode is a computationally expensive operation and might not work properly on some devices.
+    // Another reason would be the lack of auto-focus producing blurry images which makes the localization feature very unstable.
+    // However, even if none of the above apply, there is one more case where it might be useful to disable locate: If the orientation, and/or the approximate position of the barcode is known,
+    // or if you want to guide the user through a rectangular outline. This can increase performance and robustness at the same time."
+    locate: false,
```

Comment:
Some config improvement: from the doc: `In cases where the barcode is further away from the camera lens (lack of auto-focus, or small barcodes) then it's advised to set the size to small`
Book barcodes a generally quite small, ad some cameras cannot focus enough if pu to close (to fill the green box), i think we should give it a shot. When tested locally, it did improve how many barcodes were recognized 
```suggestion
    locate: false,
    locator: {
      patchSize: "small"
    }
```

Reply:
This makes close scans fail, so that would require decreasing the green box to let users have a clue on the smaller expected size, or a way to detect if the camera can autofocus. I would prefer to leave that kind of adjustments for later.